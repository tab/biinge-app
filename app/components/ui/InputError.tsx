import React from "react"
import { View, Text } from "react-native"

import { inputErrorStyles } from "styles"

type Props = {
  children: React.ReactNode
}

const InputErrorComponent = ({ children }: Props) => {
  return (
    <>
      {children && (
        <View style={inputErrorStyles.container}>
          <Text style={inputErrorStyles.text}>{children}</Text>
        </View>
      )}
    </>
  )
}

export default InputErrorComponent
