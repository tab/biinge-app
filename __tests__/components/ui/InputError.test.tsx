import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"

import { it } from "@jest/globals"

import InputError from "components/ui/InputError"

describe("InputError", () => {
  const error = "Error message"

  it("renders correctly", () => {
    const tree = render(<InputError>{error}</InputError>)
    expect(tree).toMatchSnapshot()
  })
})
