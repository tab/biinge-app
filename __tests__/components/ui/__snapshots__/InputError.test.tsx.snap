// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
      "marginTop": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Error message
  </Text>
</View>
`;
